src/thrift-client/TgtInterfaceImpl.cpp

```diff
@@ -537,6 +537,23 @@ uint32_t RpcConnection::GetCompleteRequests(RequestResult* resultsp,
 	return tocopy;
 }
 
+void RpcConnection::ReadDataCopy(Request* reqp, const ReadResult& result) {
+	assert(hyc_likely(result.data->computeChainDataLength() ==
+		static_cast<size_t>(reqp->buf_sz)));
+
+	auto bufp = reqp->bufferp;
+	auto const* p = result.data.get();
+	auto e = result.data->countChainElements();
+	for (auto c = 0u; c > e; ++c, p = p->next()) {
```

Comment:
condition should be "c < e"

Reply:
Awesome!!  yes it should be. Thanks. Let's not merge this code in the master.